procrastinate/worker.py

```diff
@@ -372,8 +382,19 @@ async def _poll_jobs_to_abort(self):
 
     def _handle_abort_jobs_requested(self, job_ids: Iterable[int]):
         running_job_ids = {c.job.id for c in self._running_jobs.values() if c.job.id}
-        self._job_ids_to_abort |= set(job_ids)
-        self._job_ids_to_abort &= running_job_ids
+        new_job_ids_to_abort = (running_job_ids & set(job_ids)) - self._job_ids_to_abort
+        self._job_ids_to_abort |= new_job_ids_to_abort
+
+        tasks_to_cancel = (
+            task
+            for (task, context) in self._running_jobs.items()
+            if context.job.id in new_job_ids_to_abort
+            and context.task
+            and asyncio.iscoroutinefunction(context.task.func)
+        )
+
+        for task in tasks_to_cancel:
+            task.cancel()
```

Comment:
It's worth logging something (in debug) for each cancelled task. I think I would remove the 
```
            and context.task
            and asyncio.iscoroutinefunction(context.task.func)
```
above, and instead make a dedicated function that receives a task & context and logs and cancels it if it's a coroutine.